src/gui/folderman.cpp

```diff
@@ -1220,7 +1220,28 @@ QString FolderMan::trayTooltipStatusString(
     return folderMessage;
 }
 
-static QString checkPathValidityRecursive(const QString &path)
+// QFileInfo::canonicalPath returns an empty string if the file does not exist.
+// This function also works with files that does not exist and resolve the symlinks in the
+// parent directories.
+static QString canonicalPath(const QString &path)
```

Comment:
maybe move it to anonymous namespace?

Reply:
I have not touched this method, just moved `checkPathValidityRecursive` below it (to have class methods next to each other).